pkg/clustermanager/retrier_client.go

```diff
@@ -23,6 +24,31 @@ func NewRetrierClient(client *client, retrier *retrier.Retrier) *retrierClient {
 	}
 }
 
+func (c *retrierClient) installEksdComponents(ctx context.Context, clusterSpec *cluster.Spec, cluster *types.Cluster) error {
+	eksdComponents, err := clusterSpec.ReadEksdManifests(clusterSpec.VersionsBundle.EksD)
+	if err != nil {
+		return fmt.Errorf("failed loading manifest for eksd components: %v", err)
+	}
+
+	if err = c.Retry(
+		func() error {
+			return c.ApplyKubeSpecFromBytesWithNamespace(ctx, cluster, eksdComponents.ReleaseCrdContent, constants.EksaSystemNamespace)
```

Comment:
I wonder if this should be part of the "upgrade core components" or not. wdy?

Reply:
I wondered the same thing. Since this method is called in both create and upgrade, if there's an update to the eksd manifest in the bundle, it'll be applied to the cluster here. So i don't think it's necessary to add it to the core components upgrades

Reply:
Oh, I'm not saying it is necessary. But it's more about concern separation.
The CRDs for eks-a get updated in the core component upgrades. And then we later apply the changes to the eks-a objects.

I was wondering if we should follow the same pattern here: upgrade the CRDs in core component upgrades and then later after the cluster upgrade has been done, create the new ekd-d release object (if there is a new one).

Reply:
#1436